include/linux/list.h

```diff
@@ -362,6 +362,17 @@ static inline void list_splice_tail_init(struct list_head *list,
 	list_entry((ptr)->next, type, member)
 
 /**
+ *  * list_last_entry - get the last element from a list
+ *   * @ptr:	the list head to take the element from.
+ *    * @type:	the type of the struct this is embedded in.
+ *     * @member:	the name of the list_struct within the struct.
+ *      *
+ *       * Note, that list is expected to be not empty.
+ *        */
```

Comment:
I found the do_mmap_pgoff() but can't found the do_mmap() in /mm/maps.

And When Pagefault happened ,which function will be called and where is this function?
I just found ——vm_operations_struct  in linux/mm.h (int (*fault)(struct vm_area_struct *vma, struct vm_fault *vmf);)

If I use the DRAM to simulate the NVM, I want to mark NVM use the vm_flag, and when I meet Pagefault alloc NVM (analog with DRAM) to the VMA with the flag bit as NVM.  
how do I need to modify the mmap in the kernel?